Cargo.toml

```
@@ -33,6 +33,11 @@ allocator-api2 = { version = "0.2.9", optional = true, default-features = false,
 # NB: this is a public dependency because `Equivalent` is re-exported!
 equivalent = { version = "1.0", optional = true, default-features = false }
 
+# serde v1.0.220 is the first version that released with `serde_core`.
+# This is required to avoid conflict with other `serde` users which may require an older version.
+[target.'cfg(any())'.dependencies]
+serde = { version = "1.0.220", default-features = false }
```

I think this hack should follow the `optional` state, so it has no effect otherwise.
```suggestion
serde = { version = "1.0.220", default-features = false, optional = true }
```